modules/drivers/mongo/src/metabase/driver/mongo/util.clj

```diff
@@ -89,35 +90,119 @@
+          `mg/connect`.  The fix was to create a connection string and use DNS SRV for
+          FQDNS.  In this fn we provide the correct connection fn based on host."
+          {:arglists '([connection-info])}
+          :type)
+
+(defmethod connect :srv
+  [{:keys [^MongoClientURI uri ]}]
+  (let [mongo-client (MongoClient. uri)]
+    (if-let [db-name (.getDatabase uri)]
+      [mongo-client (.getDB mongo-client db-name)]
+      (throw (ex-info (str (tru "No database name specified in URI. Monger requires a database to be explicitly configured." ))
+                      {:hosts (-> uri .getHosts)
+                       :uri   (-> uri .getURI)
+                       :opts  (-> uri .getOptions)})))))
+
+(defmethod connect :normal
+  [{:keys [server-address options credentials dbname]}]
+  (let [do-connect (partial mg/connect server-address options)
+        mongo-client (if credentials
+                       (do-connect credentials)
+                       (do-connect))]
+    [mongo-client (mg/get-db mongo-client dbname)]))
+
+
 (defn -with-mongo-connection
-  "Run F with a new connection (bound to `*mongo-connection*`) to DATABASE.
-   Don't use this directly; use `with-mongo-connection`."
+  "Run `f` with a new connection (bound to `*mongo-connection*`) to `database`. Don't use this directly; use
+  `with-mongo-connection`."
   [f database]
   (let [details (database->details database)]
     (ssh/with-ssh-tunnel [details-with-tunnel details]
-      (let [{:keys [dbname host port user pass ssl authdb tunnel-host tunnel-user tunnel-pass additional-options]
-             :or   {port 27017, pass "", ssl false}} details-with-tunnel
-            user             (when (seq user) ; ignore empty :user and :pass strings
-                               user)
-            pass             (when (seq pass)
-                               pass)
-            authdb           (if (seq authdb)
-                               authdb
-                               dbname)
-            server-address   (mg/server-address host port)
-            credentials      (when user
-                               (mcred/create user authdb pass))
-            connect          (partial mg/connect server-address (build-connection-options :ssl? ssl, :additional-options additional-options))
-            conn             (if credentials
-                               (connect credentials)
-                               (connect))
-            mongo-connection (mg/get-db conn dbname)]
-        (log/debug (u/format-color 'cyan "<< OPENED NEW MONGODB CONNECTION >>"))
-        (try
-          (binding [*mongo-connection* mongo-connection]
-            (f *mongo-connection*))
-          (finally        (mg/disconnect conn)
-                          (log/debug (u/format-color 'cyan "<< CLOSED MONGODB CONNECTION >>"))))))))
+                         (let [connection-info   (details->mongo-connection-info (normalize-details details-with-tunnel))
```

Comment:
I don't know why the old code was indented that way but the same rule https://guide.clojure.style/#body-indentation applies here. The body of `finally` should be indented 2 spaces (hopefully your editor does this for you automatically)

Reply:
doh, having trouble getting intellij to use .editorconfig though in theory should work

Reply:
Sorry. Ignore the last comment. It's not the body of a `finally` form that is misindented but rather the body of the `ssh/with-ssh-tunnel` macro. The same https://guide.clojure.style/#body-indentation rule still applies

Reply:
Fixed